app/controllers/api/encryptables_controller.rb

```diff
@@ -68,7 +68,7 @@ def model_class
   end
 
   def define_model_class
-    if credential_id.present?
+    if credential_id.present? || params[:folder_id].present?
```

Comment:
interesting ... why does this work with the folder_id?
doesn't have a credential also have a folder_id ?

I would say this check is rather a bit shaky .. is there a more solid way to determine the model class?

Reply:
No, the attachment just have a credential_id not a folder id. Since we don't have any useful params the check via folderId is the best solution.